pkg/registry/apps/rest/storage_apps.go

```diff
@@ -63,9 +63,13 @@ func (p RESTStorageProvider) v1beta1Storage(apiResourceConfigSource serverstorag
 		storage["deployments/scale"] = deploymentStorage.Scale
 	}
 	if apiResourceConfigSource.ResourceEnabled(version.WithResource("statefulsets")) {
-		statefulsetStorage, statefulsetStatusStorage := statefulsetstore.NewREST(restOptionsGetter)
-		storage["statefulsets"] = statefulsetStorage
-		storage["statefulsets/status"] = statefulsetStatusStorage
+		// enable scale subresource for v1beta1 for e2e testing
+		// we are not able to test v1beta2 as v1beta2 is disabled at master.go
+		// TODO (juntee): remove the subresource before 1.8 release
```

Comment:
IMO there is no reason to add a new subresource to a version to be retired soon. The only reason we add the /scale for v1beta1 is for e2e testing purpose. If having integration test in this PR is better, we should not add /scale to v1beta1.
@kow3ns @foxish